DemoApp/Sources/Components/AppState.swift

```diff
@@ -20,8 +20,24 @@ final class AppState: ObservableObject {
     @Published var loading = false
     @Published var activeCall: Call?
     @Published var activeAnonymousCallId: String = ""
-    @Published var voIPPushToken: String? { didSet { didSet(voIPPushToken: voIPPushToken) } }
-    @Published var pushToken: String? { didSet { didSet(pushToken: pushToken) } }
+    @Published var voIPPushToken: String? {
+        didSet {
+            if voIPPushToken != oldValue {
+                unsecureRepository.save(voIPPushToken: voIPPushToken)
+                didSet(voIPPushToken: voIPPushToken)
```

Comment:
I was a bit confused how `didSet` can be called within `didSet` but it was actually a private method. `commit/apply(voIPPushToken)` could possibly clearer. 
Feel free to discard this comment.